discovery-provider/src/solana/anchor_program_indexer.py

```diff
@@ -132,5 +145,49 @@ def process_index_task(self):
     # TODO - Override with actual remote fetch operation
     # parsed_transactions will contain an array of txs w/instructions
     # each containing relevant metadata in container
-    async def fetch_ipfs_metadata(self, parsed_transactions):
-        return super().fetch_ipfs_metadata(parsed_transactions)
+    async def fetch_ipfs_metadata(
+        self, parsed_transactions: List[Dict]
+    ) -> Tuple[Dict[str, Dict], Set[str]]:
+
+        cid_to_user_id: Dict[str, int] = {}
+        cids_txhash_set: Set[Tuple[str, str]] = set()
```

Comment:
jw why is this a set? is it possible to have non-unique mappings here that we want to control for?

Reply:
shouldn't be but just in case, there's no harm in using a set.